rfcs/active/0000-badge.md

```diff
@@ -0,0 +1,362 @@
+* Start Date: 2025-02-24
+* RFC PR: (Leave empty till PR Created)
+
+# Summary
+
+This RFC proposes the addition of a Badge component to the OpenUI Design System. Badges are small, visually distinct elements used to display counts, status indicators, or short labels. They're commonly used in interfaces to highlight new content, show notification counts, indicate status, and provide visual cues for users. Badges typically appear in proximity to notifications or user avatars with eye-catching appeal, helping users quickly identify important information.
+
+# Basic Example
+
+```html
+<!-- Basic badge -->
+<oui-badge>New</oui-badge>
+
+<!-- Badge with variant -->
+<oui-badge variant="primary">4</oui-badge>
+
+<!-- Pill badge -->
+<oui-badge shape="pill">New</oui-badge>
+
+<!-- Badge in context with another element -->
+<h2>Product Features <oui-badge>New</oui-badge></h2>
+
+<!-- Badge as a counter -->
+<oui-button>
+  Notifications
+  <oui-badge variant="secondary">4</oui-badge>
+</oui-button>
+
+<!-- Positioned badge (for notification indicators) -->
+<oui-button>
+  Inbox
+  <oui-badge position="top-end" variant="danger">99+</oui-badge>
+</oui-button>
+
+<!-- Dot badge (for status indicators) -->
+<oui-badge dot aria-label="New notifications"></oui-badge>
+
+<!-- Status badge -->
+<oui-badge status="success">Success</oui-badge>
+
+<!-- Badge with custom color -->
+<oui-badge color="#52c41a">Custom</oui-badge>
+
+<!-- Badge with overflow count -->
+<oui-badge count="125" overflow-count="99"></oui-badge>
+
+<!-- Standalone ribbon badge -->
+<oui-badge.ribbon text="Ribbon" placement="end">
+  <div style="width: 300px; height: 100px; background: #f0f0f0;"></div>
+</oui-badge.ribbon>
+```
+
+# Motivation
+
+Badges are a fundamental UI pattern found in many digital interfaces across various platforms. They serve several key purposes:
+
+1. **Information density** - Badges allow developers to display additional information (counts, status, labels) in a compact space.
+2. **Visual indication** - They provide immediate visual cues to users about state, notifications, or important updates.
+3. **Context enhancement** - They enhance other components by providing supplementary information (counters, status indicators).
+
+The OpenUI Design System aims to provide a comprehensive set of common UI components, and badges represent a widely-used pattern that fits within this mission. Adding a standardized badge component would reduce duplicative effort in rebuilding this common UI pattern across different applications and frameworks.
+
+# Detailed Proposal
+
+The `oui-badge` component will provide a versatile, accessible, and themeable implementation of the badge pattern.
+
+## API
+
+### Properties
+
+| Property | Type | Default | Description |
```

Comment:
Most of these properties seem to control styling / appearance. In general, it's pretty limiting to set these directly on the element v. via css: (1) you need access to the element via HTML, (2) CSS is the standard tool for styling appearance, (3) CSS is highly configurable via the cascade and inheritance.

I think the desired functionality here could be implemented by using custom properties and style container queries.

Reply:
This [technique](https://stackblitz.com/edit/vitejs-vite-ltjx4uyb?file=index.html) can be used to enable arbitrary styling based on custom properties matching.